worlds/satisfactory/__init__.py

```diff
@@ -0,0 +1,264 @@
+        if self.options.splitter_placement == Placement.starting_inventory:
+            self.push_precollected_by_name("Building: Conveyor Splitter")
+            self.push_precollected_by_name("Building: Conveyor Merger")
+
+        if not self.options.trap_selection_override.value:
+            self.options.trap_selection_override.value = set(self.options.trap_selection_preset.get_selected_list())
+
+        self.critical_path = CriticalPathCalculator(self.game_logic, self.critical_path_seed, self.options)
+        self.critical_path.calculate()
+
+        self.state_logic = StateLogic(self.player, self.options, self.critical_path)
+        self.items = Items(self.player, self.game_logic, self.random, self.options, self.critical_path)
+
+        starting_inventory: list[str] = self.options.starting_inventory_preset.get_selected_list()
+        for item_name in starting_inventory:
+            self.push_precollected_by_name(item_name)
+
+    def create_regions(self) -> None:
+        locations: list[LocationData] = \
+            Locations(self.game_logic, self.options, self.state_logic, self.items, self.critical_path).get_locations()
+        create_regions_and_return_locations(
+            self.multiworld, self.options, self.player, self.game_logic, self.state_logic, self.critical_path,
+            locations)
+
+    def create_items(self) -> None:
+        self.setup_events()
+
+        number_of_locations: int = len(self.multiworld.get_unfilled_locations(self.player))
+        precollected_items: list[Item] = self.multiworld.precollected_items[self.player]
+
+        self.multiworld.itempool += \
+            self.items.build_item_pool(self.random, precollected_items, number_of_locations)
+
+    def set_rules(self) -> None:
+        resource_sink_goal: bool = "AWESOME Sink Points (total)" in self.options.goal_selection \
+                                   or "AWESOME Sink Points (per minute)" in self.options.goal_selection
+
+        required_parts = set(self.game_logic.space_elevator_phases[self.options.final_elevator_phase.value - 1].keys())
+
+        if resource_sink_goal:
+            required_parts.union(self.game_logic.buildings["AWESOME Sink"].inputs)
+
+        self.multiworld.completion_condition[self.player] = \
+            lambda state: self.state_logic.can_produce_all(state, required_parts)
+
+    def collect(self, state: CollectionState, item: Item) -> bool:
+        change = super().collect(state, item)
+        if change and item.name in self.game_logic.indirect_recipes:
+            state.prog_items[self.player][self.game_logic.indirect_recipes[item.name]] = 1
+        return change
+
+    def remove(self, state: CollectionState, item: Item) -> bool:
+        change = super().remove(state, item)
+        if change and item.name in self.game_logic.indirect_recipes:
+            del state.prog_items[self.player][self.game_logic.indirect_recipes[item.name]]
```

Comment:
There are features in AP that can add additional copies of items, such as plando with from_pool: false, or item link replacement. This `= 1` and `del` thing won't work correctly in those cases. Just use += 1 and -= 1, no need for del. Core knows how to handle that and that's what other worlds do.